File: CourseProjects/CourseProjects/VisualStudioProjects/Agile/LinkedListChallenge/LinkedListMSTestStart/LinkedList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace LinkedListMSTestStart {
    public class LinkedList {
        public bool IsEmpty {
            get { return size == 0; }
        }
        public int Size {
            get { return size; }
        }
        public void Add(string item) {
            if (IsEmpty) {
                first = new Node(null, null, item);
            }
            size++;
        }
        public string this[int index] {
            get {
                return Get(index);
            }
        }
        private string Get(int index) {
            if (index == 0) {
                return first.Item;
            }
            return null;
        }
        private int size;
        private Node first;
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/JavaLanguageExercises/src/linkedlist/start/LinkedList.java
package linkedlist.start;

public class LinkedList {

	public boolean isEmpty() {
		return size() == 0;
	}

	public int size() {
		return size;
	}

	public void add(String s) {
		if(isEmpty()) {
			first = new Node(s,null,null);
		}
		size++;
	}
	public String get(int i) {
		if(i == 0) {
			return first.getItem();
		}
		return null;
	}
	private int size;
	private Node first;
}
